src/languages/clojure/index.ts

```diff
@@ -0,0 +1,196 @@
+import {
+  type Language,
+  required,
+  simplegolf,
+  search,
+} from "../../common/Language";
+import { ClojureEmitter } from "./emit";
+import {
+  arraysToLists,
+  flipBinaryOps,
+  flipped,
+  mapBackwardsIndexToForwards,
+  mapOps,
+  mapOpsTo,
+  removeImplicitConversions,
+} from "../../plugins/ops";
+import {
+  builtin,
+  conditional,
+  functionCall as func,
+  int,
+  op,
+  text,
+  isInt,
+  intToDecOpOrText,
+} from "../../IR";
+import {
+  golfLastPrint,
+  golfLastPrintInt,
+  putcToPrintChar,
+} from "../../plugins/print";
+import { usePrimaryTextOps } from "../../plugins/textOps";
+import { golfStringListLiteral, listOpsToTextOps } from "../../plugins/static";
+import {
+  applyDeMorgans,
+  bitnotPlugins,
+  equalityToInequality,
+  lowBitsPlugins,
+  pickAnyInt,
+  truncatingOpsPlugins,
+} from "../../plugins/arithmetic";
+import { forArgvToForEach } from "../../plugins/loops";
+import { alias, renameIdents } from "../../plugins/idents";
+import { assertInt64 } from "../../plugins/types";
+import { implicitlyConvertConcatArg } from "./plugins";
+
+const clojureLanguage: Language = {
+  name: "Clojure",
+  extension: "clj",
+  emitter: new ClojureEmitter(),
+  phases: [
+    required(arraysToLists, putcToPrintChar, usePrimaryTextOps("codepoint")),
+    simplegolf(golfLastPrint(false), golfLastPrintInt(false)),
+    search(
+      flipBinaryOps,
+      golfStringListLiteral(false),
+      listOpsToTextOps(),
+      equalityToInequality,
+      ...bitnotPlugins,
+      ...lowBitsPlugins,
+      applyDeMorgans,
+    ),
+
+    required(
+      pickAnyInt,
+      forArgvToForEach,
+      mapOps({
+        right_align: (a, b) =>
+          func(
+            "format",
+            op["concat[Text]"](text("%"), intToDecOpOrText(b), text("s")),
+            a,
+          ),
+        int_to_hex_aligned: (a, b) =>
+          func(
+            "format",
+            op["concat[Text]"](text("%0"), intToDecOpOrText(b), text("x")),
+            a,
+          ),
+        int_to_Hex_aligned: (a, b) =>
+          func(
+            "format",
+            op["concat[Text]"](text("%0"), intToDecOpOrText(b), text("X")),
+            a,
+          ),
+      }),
+    ),
+    simplegolf(implicitlyConvertConcatArg),
+    required(
```

Comment:
There are some opcodes that could be mapped using `mapOpsTo.func` instead of the general `mapOps`.